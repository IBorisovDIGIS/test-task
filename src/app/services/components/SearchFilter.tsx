import React, { ChangeEvent } from 'react';
import {
  Button, Grid, makeStyles, TextField,
} from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme) => ({
  margin: {
    margin: theme.spacing(1),
  },
  formControl: {
    flex: 1,
  },
  inputLabel: {
    position: 'relative',
    top: -25,
    padding: 0,
    margin: 0,
  },
  input: {
    borderRadius: 4,
    backgroundColor: theme.palette.common.white,
    border: '1px solid #ced4da',
    fontSize: 16,
    padding: '5px 10px',
  },
  button: {
    padding: '8px 10px',
  },
}));

interface ISearchFilter {
    handleChangeSearch: (value: string) => void;
    searchValue: string;
}

const SearchFilter: React.FC<ISearchFilter> = ({
  handleChangeSearch,
  searchValue,
}: ISearchFilter) => {
  const classes = useStyles();
  const { t } = useTranslation();

  const handleChangeFilter = (event: ChangeEvent<HTMLInputElement>) => {
    handleChangeSearch(event.target.value);
  };

  const handleResetFilter = () => handleChangeSearch('');

  return (
    <Grid container spacing={1}>
      <Grid item xs={4} container alignItems="stretch">
        <TextField
          fullWidth
          label={t('filter')}
          InputProps={{ disableUnderline: true, classes: { root: classes.input } }}
          InputLabelProps={{
            shrink: true,
          }}
          value={searchValue}
          onChange={handleChangeFilter}
        />
      </Grid>
      <Grid item xs={4} container alignItems="flex-end">
        <Button
          variant="outlined"
          classes={{ root: classes.button }}
          onClick={handleResetFilter}
        >
          {t('reset')}
        </Button>
      </Grid>
    </Grid>
  );
};

export default SearchFilter;
