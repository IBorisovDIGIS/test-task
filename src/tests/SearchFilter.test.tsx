import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  Button,
  TextField,
} from '@material-ui/core';
import toJson from 'enzyme-to-json';
import SearchFilter from "../app/services/components/SearchFilter";

configure({ adapter: new Adapter() });

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key })
}));

describe('Search filter should render correctly', () => {
  it('Should renders correctly', () => {
    const handleChangeSearch = jest.fn();

    const container = shallow(<SearchFilter handleChangeSearch={handleChangeSearch} searchValue="" />);

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should change search text', async () => {
    const handleChangeSearch = jest.fn();
    const container = shallow(<SearchFilter handleChangeSearch={handleChangeSearch} searchValue="" />);
    container.find(TextField).simulate('change', {
      target: { value: "test123" }
    });
    expect(handleChangeSearch).toHaveBeenCalledWith('test123')
  });

  it('Should render search value correctly', async () => {
    const handleChangeSearch = jest.fn();
    const container = shallow(<SearchFilter handleChangeSearch={handleChangeSearch} searchValue="test" />);
    expect(container.find(TextField).prop('value')).toEqual('test')
  });

  it('Should reset value on by clicking on reset button', async () => {
    const handleChangeSearch = jest.fn();
    const container = shallow(<SearchFilter handleChangeSearch={handleChangeSearch} searchValue="" />);
    container.find(Button).simulate('click');
    expect(handleChangeSearch).toHaveBeenCalledWith('')
  });
});
