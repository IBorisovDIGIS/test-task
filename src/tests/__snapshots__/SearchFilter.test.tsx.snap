// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search filter should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    item={true}
    xs={4}
  >
    <WithStyles(ForwardRef(TextField))
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      InputProps={
        Object {
          "classes": Object {
            "root": "makeStyles-input-4",
          },
          "disableUnderline": true,
        }
      }
      fullWidth={true}
      label="filter"
      onChange={[Function]}
      value=""
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    item={true}
    xs={4}
  >
    <WithStyles(ForwardRef(Button))
      classes={
        Object {
          "root": "makeStyles-button-5",
        }
      }
      onClick={[Function]}
      variant="outlined"
    >
      reset
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
